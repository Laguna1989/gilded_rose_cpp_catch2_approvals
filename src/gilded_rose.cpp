#include "gilded_rose.hpp"

#include <items/item_converter.hpp>

GildedRose::GildedRose(std::vector<Item>& items)
{
    internalItems = convert(items);
}

void GildedRose::updateQuality()
{
    for (auto& item : internalItems) {
        item->update();
    }
}
