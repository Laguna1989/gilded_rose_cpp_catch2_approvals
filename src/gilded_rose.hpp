#include <string>
#include <vector>

class Item {
public:
    std::string name;
    int sellIn;
    int quality;

    Item(std::string const& name, int sellIn, int quality)
        : name(name)
        , sellIn(sellIn)
        , quality(quality)
    {
    }
};

class GildedRose {
public:
    explicit GildedRose(std::vector<Item>& items);

    void updateQuality();

private:
    std::vector<Item>& items;
};
