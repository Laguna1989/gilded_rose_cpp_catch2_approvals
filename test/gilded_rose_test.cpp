#include "gilded_rose.hpp"

#include <ApprovalTests.hpp>
#include <catch2/catch_test_macros.hpp>

std::ostream& operator<<(std::ostream& os, Item const& obj)
{
    return os << "name: " << obj.name << ", sellIn: " << obj.sellIn
              << ", quality: " << obj.quality;
}

TEST_CASE("Single item")
{
    std::vector<Item> items {Item{"Axe", 10, 15}};
    GildedRose gildedRose{items};

    gildedRose.updateQuality();

    ApprovalTests::Approvals::verify(items[0]);
}

TEST_CASE("Multiple items")
{
    std::vector<Item> items {Item{"Axe", 10, 15}, Item{"Backstage passes to a TAFKAL80ETC concert", 10, 15}};
    GildedRose gildedRose{items};

    gildedRose.updateQuality();

    ApprovalTests::Approvals::verifyAll(items);
}
