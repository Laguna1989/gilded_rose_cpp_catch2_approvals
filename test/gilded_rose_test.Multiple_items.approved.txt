[0] = name: Axe, sellIn: 9, quality: 14
[1] = name: Backstage passes to a TAFKAL80ETC concert, sellIn: 9, quality: 17

